### PR TITLE
Feat: added rlc tests

### PR DESCRIPTION
**Pull request by rpalakkal**

### CODE REVIEW

File: src/rlp/tests.rs [hunk 1]
```diff
@@ -81,6 +85,106 @@ mod rlc {
...
+        b.resize(combined_len as usize, F::from(0));
+        let b = ctx.assign_witnesses(b);
+
+        let gate = GateChip::default();
+        let (ctx_gate, ctx_rlc) = rlc_builder.rlc_ctx_pair();
+        let rlc = RlcChip::new(F::from(2));
+        let rlc_trace = rlc.compute_rlc((ctx_gate, ctx_rlc), &gate, a_b, a_b_len);
+        let rlc_val = *rlc_trace.rlc_val.value();
+
+        let rlc_trace_a = rlc.compute_rlc((ctx_gate, ctx_rlc), &gate, a, a_len);
```
Comment (yi-sun) [inline src/rlp/tests.rs]: Should also test inputs where `a_len` is not equal to `a.len()`.  This probably requires adding additional test case inputs for `a_len`, `b_len`, and `a_b_len`.

Comment (yi-sun) [reply src/rlp/tests.rs]: Will also be interesting to add edge cases for `a_len = 0` or `b_len = 0`.

**Review by yi-sun — CHANGES_REQUESTED** [review]

File: src/rlp/tests.rs [hunk 2]
```diff
@@ -81,6 +85,106 @@ mod rlc {
...
+        let rlc_val_a = *rlc_trace_a.rlc_val.value();
+
+        let rlc_trace_b = rlc.compute_rlc((ctx_gate, ctx_rlc), &gate, b, b_len);
+        let rlc_val_b = *rlc_trace_b.rlc_val.value();
+
+        rlc.load_rlc_cache((ctx_gate, ctx_rlc), &gate, combined_len as usize);
+        let gamma_pow = rlc.rlc_pow(ctx_gate, &gate, b_len, bit_length(combined_len as u64));
+        assert_eq!(rlc_val, rlc_val_a * gamma_pow.value() + rlc_val_b);
+
```
Comment (yi-sun) [inline src/rlp/tests.rs]: Should also check the actual values of `rlc_val`, `rlc_val_a`, and `rlc_val_b` against an external computation with the mocked challenge value.

File: src/rlp/tests.rs (hunk 1 above)
Comment (yi-sun) [reply src/rlp/tests.rs]: Would be nice to have some test cases where `a_len < a.len()` and `b_len < b.len()` but `a_len + b_len = a_b_len`.  OK to merge after adding.

**Review by yi-sun — APPROVED** [review]

**Comment by yi-sun** [comment]
LGTM!
